### PR TITLE
Added the app

### PR DESCRIPTION
**Pull request by UKnowWhoIm**
- Works perfectly when run as a JS script locally
- Encounters "Resource not accessible by integration"(Code 403) error if triggered by a workflow

### CODE REVIEW

**Review by gautamkrishnar — CHANGES_REQUESTED** [review]
**Please remove node modules**, and add it to .gitignore

**Comment by gautamkrishnar** [comment]
@UKnowWhoIm You can also use the default node gitignore file for your gitignore https://github.com/github/gitignore/blob/master/Node.gitignore

**Comment by UKnowWhoIm** [comment]
Made the changes

File: index.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+const orgName = core.getInput('org-name');
+const authToken = process.env["GITHUB_TOKEN"];
+const path = core.getInput("file-path");
+const name = core.getInput("commit-user-name");
+const email = core.getInput("commit-user-email");
+const message = core.getInput("commit-msg")
+// Eliminate Owner name from repo;
+const repo = process.env["GITHUB_REPOSITORY"].split("/")[1]
+
```
Comment (gautamkrishnar) [inline index.js:15]: create a new var named `repo_owner` here.
```
const repo_owner= process.env["GITHUB_REPOSITORY"].split("/")[0];
```

File: index.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        teams_response =  await octokit.teams.list({
+            org: orgName,
+        }).catch(
+            err => console.log(err)
+        );
+        
+        member_data = await getMemberData(teams_response.data);
+        
+        json_file_response = await octokit.repos.getContent({
+            owner: orgName,
```
Comment (gautamkrishnar) [inline index.js]: Use `repo_owner` here instead of using orgName.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            repo: repo,
+            path: path,
+        }).catch(
+            err => console.log(err)
```
Comment (gautamkrishnar) [inline index.js]: You can exit the script here (`process.exit(1)`) if the file API goes wrong. Print error using `core.error(e)`

File: index.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        if (!authToken) {
+            throw new Error("Token not found");
+        }
+
+        octokit = github.getOctokit(authToken);
+
+        teams_response =  await octokit.teams.list({
+            org: orgName,
+        }).catch(
+            err => console.log(err)
```
Comment (gautamkrishnar) [inline index.js]: Here also exit with 1 if error

File: index.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                committer: {
+                    name: name,
+                    email: email
+                }
+            })
+            .then(  
+                core.setOutput("message", "File Updated"), console.log("file updated")
+            )
+            .catch(
+                err => console.log(err)
```
Comment (gautamkrishnar) [inline index.js]: exit 1 if error

**Review by gautamkrishnar — CHANGES_REQUESTED** [review]

**Comment by gautamkrishnar** [comment]
@UKnowWhoIm please make all the requested changes

File: index.js (hunk 1 above)
Comment (gautamkrishnar) [reply index.js:15]: @UKnowWhoIm you missed this

File: index.js (hunk 3 above)
Comment (gautamkrishnar) [reply index.js]: @UKnowWhoIm and this

File: index.js (hunk 1 above)
Comment (UKnowWhoIm) [reply index.js:15]: Nope i changed it to
`const [repo_owner, repo_name] = process.env["GITHUB_REPOSITORY"].split("/")`
Idk why this is still showing the old code, Checkout line 14 of `[index.js](https://github.com/UKnowWhoIm/members/blob/main/index.js)

File: index.js (hunk 3 above)
Comment (UKnowWhoIm) [reply index.js]: I changes this too [index.js](https://github.com/UKnowWhoIm/members/blob/main/index.js) @ line 78

File: index.js (hunk 1 above)
Comment (UKnowWhoIm) [reply index.js:15]: @gautamkrishnar

**Comment by UKnowWhoIm** [comment]
@gautamkrishnar I made the changes, but they're only visible if you click Files Changed **NOT** View Changes. Idk why :|

**Review by gautamkrishnar — APPROVED** [review]

**Pull request merged**